tests/sweep_framework/sweeps/all_gather_T3K_sharded.py

```diff
@@ -0,0 +1,239 @@
+    unchunked_input_shape = list(test_vector["input_shape"])
+    unchunked_input_shape[test_vector["dim"]] *= test_vector["num_devices"]
+    if test_vector["num_devices"] < 2:
+        return True, f"Requires multiple devices to run"
+    elif test_vector["num_devices"] == 2 and test_vector["num_links"] == 2:
+        return True, f"Not enough links to run"
+
+    if unchunked_input_shape[test_vector["dim"]] % test_vector["num_devices"] != 0 or (
+        test_vector["dim"] == 3 and unchunked_input_shape[test_vector["dim"]] // test_vector["num_devices"] % 32 != 0
+    ):
+        return True, f"Unsupported test case"
+    if test_vector["tensor_layout"] == ttnn.ROW_MAJOR_LAYOUT and test_vector["input_dtype"] == ttnn.bfloat8_b:
+        return True, f"bfloat8_b/4_b only supports TILE layout"
+    if test_vector["tensor_layout"] == ttnn.ROW_MAJOR_LAYOUT:
+        return True, f"ROW_MAJOR_LAYOUT not supported"
+    if test_vector["tensor_mem_layout"] == ttnn.TensorMemoryLayout.WIDTH_SHARDED:
+        return True, f"Output mismatch"
+    if test_vector["num_links"] == 2 and test_vector["num_devices"] == 8:
+        return True, f"8 devices and 2 links are not supported"
+    if test_vector["input_shard_shape"][1] != test_vector["input_shape"][-1]:
+        return True, f"Shard width must be equal to last dim of shape"
+    if test_vector["tensor_mem_layout"] != ttnn.TensorMemoryLayout.BLOCK_SHARDED and test_vector["dim"] != 3:
+        return True, "BLOCK_SHARDED is only supported for dim = 3"
+    if test_vector["tensor_mem_layout"] == ttnn.TensorMemoryLayout.SINGLE_BANK:
+        return True, f"SINGLE_BANK is not supported"
+    return False, None
+
+
+def device_mesh_fixture():
+    import tt_lib as ttl
+
+    assert ttnn.get_num_devices() >= 8, "Not T3000!"
+    device_ids = [0, 4, 5, 1, 2, 6, 7, 3]
+    num_devices_requested = len(device_ids)
+    device_mesh = ttnn.open_device_mesh(ttnn.DeviceGrid(1, num_devices_requested), device_ids[:num_devices_requested])
+    print("ALL GATHER: Opened device mesh")
+
+    yield (device_mesh, "T3000 Mesh")
+
+    print("ALL GATHER: Closing device mesh")
+    for device in device_mesh.get_devices():
+        ttl.device.DumpDeviceProfiler(device)
+    ttnn.close_device_mesh(device_mesh)
+    del device_mesh
+
+
+# This is the run instructions for the test, defined by the developer.
+# The run function must take the above-defined parameters as inputs.
+# The runner will call this run function with each test vector, and the returned results from this function will be stored.
+
+
+def run(
+    num_devices,
+    input_shape,
+    input_shard_shape,
+    shard_grid,
+    dim,
+    num_links,
+    orientation,
+    input_dtype,
+    tensor_layout,
+    tensor_mem_layout,
+    # num_cores,
+    *,
+    device,
+):
+    all_devices = device
+
+    numel = input_shape[0] * input_shape[1] * input_shape[2] * input_shape[3] * num_devices
```

Comment:
Can we not make this call the baseline run function in test_all_gather? They should ideally have the same underlying code (even if the entrypoints differ)

Reply:
It has perf analysis in between so I kept this as a new function

Reply:
I don't see why this can't be shared.

Worst case we wrap each part of the test function into separate smaller functions and call those. Otherwise we just keep the perf time stamping but disable it for the non sweep variants for the time being.